pentasharp/Controllers/AuthenticateController.cs

```diff
@@ -1,17 +1,136 @@
 using Microsoft.AspNetCore.Mvc;
+using System.Security.Cryptography;
+using Microsoft.AspNetCore.Mvc;
+using pentasharp.Data;
+using pentasharp.Models.Entities;
+using System.Text;
+using pentasharp.ViewModel.Authenticate;
 
 namespace WebApplication1.Controllers
 {
     public class AuthenticateController : Controller
     {
-        public IActionResult Login()
+        private readonly AppDbContext _context;
+
+        public AuthenticateController(AppDbContext context)
         {
-            return View();
+            _context = context;
         }
 
         public IActionResult Register()
+        {
+            var model = new RegisterViewModel();
+            return View(model);
+        }
+
+
+        public IActionResult Login()
         {
             return View();
         }
+
+        [HttpPost]
+        public IActionResult Register(RegisterViewModel model)
+        {
+            if (ModelState.IsValid)
+            {
+                var user = new User
+                {
+                    FirstName = model.FirstName,
+                    LastName = model.LastName,
+                    Email = model.Email,
+                    PasswordHash = HashPassword(model.Password)
+                };
+
+                _context.Users.Add(user);
+                _context.SaveChanges();
+                return RedirectToAction("Index", "Home");
+            }
+            return View(model);
+        }
+
+        private string HashPassword(string password)
+        {
+            using (var sha256 = SHA256.Create())
+            {
+                var bytes = sha256.ComputeHash(Encoding.UTF8.GetBytes(password));
+                var builder = new StringBuilder();
+                foreach (var b in bytes)
+                {
+                    builder.Append(b.ToString("x2"));
+                }
+                return builder.ToString();
+            }
+        }
+        public IActionResult UserList()
+        {
+            var users = _context.Users.ToList();  
+            return View(users); 
+        }
+
+
+        public IActionResult Edit(int id)
+        {
+            var user = _context.Users.Find(id);
+            if (user == null)
+            {
+                return NotFound();
+            }
+
+            var model = new EditUserViewModel
+            {
```

Comment:
same as per previous comment:
Use AutoMapper for object mappings instead of manually assigning properties.